packages/nimble-components/src/table/specs/table-column-specs/table-column-formatted-text.md

```diff
@@ -161,21 +161,67 @@ Follows the [Column Type Philosophy](/packages/nimble-components/src/table/specs
+-   `decimal-maximum-digits` - when format is `decimal`, a number that controls the maximum number of digits shown to the right of the decimal separator. Possible values are from 0 to 20. It is invalid to specify both `decimal-digits` and `decimal-maximum-digits`. Formats other than `decimal` ignore `decimal-maximum-digits`.
 
 This column will display a blank cell when `typeof` the value is not `"number"` (i.e. if the value is `null`, `undefined`, not present, or has a different runtime data type). Note that IEE 754 numbers like Infinity, NaN, and -0 are type `"number"` so will be displayed how each formatter converts them. This will preserve values like `"∞"` and `"NaN"`.
 
-This column will trigger `invalidColumnConfiguration` on the table's validity state and will include flags in the column's validity state if its configuration can't be translated to a valid `Intl.NumberFormat` object. To provide better developer feedback about what's wrong with the configuration, the column could expose a public method like `createNumberFormat()` which would use the column's configuration to construct the formatter but allow any exceptions to be thrown.
+This column will trigger `invalidColumnConfiguration` on the table's validity state and will include flags in the column's validity state if both `decimal-digits` and `decimal-maximum-digits` are specified, or its configuration can't be translated to a valid `Intl.NumberFormat` object. To provide better developer feedback about what's wrong with the configuration, the column could expose a public method like `createNumberFormat()` which would use the column's configuration to construct the formatter but allow any exceptions to be thrown.
+
+A unit for the column may be configured by providing a `nimble-unit-<name>` element as content (in addition to the column label). Unit elements represent families of related units, e.g. `nimble-unit-byte` represents bytes, KB, MB, etc. Values are converted from a source unit (e.g. bytes) to the largest related unit (e.g. KB, MB, etc.) that can represent that value with magnitude >= 1. The source data for the column is expected to be given in the base unit specified in the tag name, e.g. for `nimble-unit-byte`, a source value should be a number of bytes. Note that unit family elements are not rendered but serve to allow selective loading of data for only the needed units. The initial set of unit elements are:
+
+-   `nimble-unit-byte` - Unit labels in this family are `byte`/`bytes`, `KB`, `MB`, `GB`, `TB`, `PB`
+-   `nimble-unit-byte-1024` - Unit labels in this family are `byte`/`bytes`, `KiB`, `MiB`, `GiB`, `TiB`, `PiB`
```

Comment:
Since we're always using child elements to pick the unit, another option instead of a dedicated element for binary byte conversions would be to use one element for both types of byte conversion and configure the behavior via an attribute on the element:
```
<nimble-unit-byte conversion="binary">
```

An obvious con of this approach is that now clients would pull in the `KiB` strings even if they're only using the `KB` unit. But otherwise I like the idea of configuring unit-specific behavior on the unit element. Some other examples (just brainstorming, not advocating these be included in the spec):

Include units that aren't powers of 1000 (see thread below)
```
<nimble-unit-meter include-centi include-deci> 
```

Pick currency
```
<nimble-unit-currnecy currency='peso'>
```

Configure "kilobytes" vs "KB"
```
<nimble-unit-byte unit-display='long'>
```

I'm personally leaning towards fewer unit elements with more configuration, even at the cost of some additional strings for the element. But open to more input or stronger opinions in other directions.

Reply:
I think your suggestion makes sense. The extra translations shouldn't be a concern, especially since I doubt we'll actually have to translate the prefixes anyway.